File: src/trackers/es/tbplus_li.ts
import {WebChecker} from "../../checker/webChecker";
import {CheckerResult} from "../../checker/checkerResult";

export default class Tbplus_li extends WebChecker {
    public get url() { return "https://tbplus.li"; }
    public get name() { return "tbplus.li"; }
    public get language() { return "es"; }
    public get categories(): Array<String> { return ["movies"]; }
    public get description() { return ""; }
    public get registryUrl() { return "https://tbplus.li/index.php?page=signup"; }

    async isOpen(): Promise<CheckerResult> {
        let $ = await WebChecker.getQueryUrl(this.registryUrl);
        let exists = $(":contains('pero los registros estan cerrados')").length > 0;

        return {
            online: true,
            regClosed: !exists,
            regOpened: false
        };
    }
}

File: checker/trackers/tbplusli.go
package trackers

import (
	"github.com/tokkenno/open-torrent-tracker-bot/checker/check"
	"github.com/tokkenno/open-torrent-tracker-bot/models"
	"strings"
)

type TbPlusLi struct {
}

func (tracker *TbPlusLi) GetId() string {
	return "tbplus_li"
}

func (tracker *TbPlusLi) GetName() string {
	return "TBPlus"
}

func (tracker *TbPlusLi) GetDescription() string {
	return ""
}

func (tracker *TbPlusLi) GetLanguages() []string {
	return []string{"es"}
}

func (tracker *TbPlusLi) GetCategories() []string {
	return []string{"movies"}
}

func (tracker *TbPlusLi) GetUrl() string {
	return "https://tbplus.li"
}

func (tracker *TbPlusLi) GetRegistryUrl() string {
	return "https://tbplus.li/index.php?page=signup"
}

func (tracker *TbPlusLi) Check() *check.Result {
	html, err := check.GetWebCode(tracker.GetRegistryUrl())

	if err != nil {
		return &check.Result{
			IsOnline: false,
		}
	} else {
		var status models.RegistrationStatus

		if strings.Contains(html, "los registros estan cerrados") {
			status = models.RegistrationClose
		} else {
			status = models.RegistrationMaybe
		}

		return &check.Result{
			IsOnline: true,
			Status:   status,
		}
	}
}
